v2/docs/OCPSTRAT-1515.md

```diff
@@ -0,0 +1,207 @@
+
+This behavior SHOULD NOT interfere with the ability to verify the catalog (or contents) signature verification, as requested in [OCPSTRAT-1417](https://issues.redhat.com/browse/OCPSTRAT-1417).
+
+Possible solutions were [discussed with Operator Framework team](https://drive.google.com/file/d/1-O1MX-tnGntrl7EMfdjekSIHsUm8TOD3/view?usp=drive_web), and the gist of the discussion shows that:
+* **For OLM-v1 catalogs**, several solutions for filtering out catalogs while preserving image's signature and SBOM are possible. A separate feature request should be created. Among the solutions mentioned:
+  * Adding a patch with filtering metadata as a [OCI referrer](https://github.com/oras-project/artifacts-spec/blob/main/manifest-referrers-api.md)
+  * Using extra fields in the cluster catalog custom resource
+* **For OLM-v0 catalogs**, no perfect solution exists that ensures signature validity at the same time. Among the solutions mentioned:
+  * no filtering: which keeps the image intact, and inline with its signature. On the downside, clusters will still suffer from [OCPBUGS-35386](https://issues.redhat.com/browse/OCPBUGS-35386).
+  * a release concept attached to the catalog image, that can be used by OLM, the webconsole, etc to patch the original FBC. CatalogSource can have extra stanza, `patches` for example. But SQLite catalogs cannot be patched. Additionally, extra effort on a soon to be deprecated version is going to be a waste.
+  * mirroring by minVersion only: Contrary to v1, oc-mirror v2 would not allow filtering by maxVersion. As a result, more operator bundles will be mirrored, up to each channel head. The catalog image is also kept intact in this solution, so inline with its signature. The clusters won't experience [OCPBUGS-35386](https://issues.redhat.com/browse/OCPBUGS-35386) as the channel head bundle is always available. The downside of this solution is that oc-mirror customers might choose not to migrate to oc-mirror v2 because it lacks the filtering by maxVersion filtering.
+  * filtering by minVersion and maxVersion: Preserving the oc-mirror v1 functionality, and rebuilding the catalog image from the filtered FBC. The downside here is that the signature verification will fail, and that the SBOM will not correspond to the filtered catalog image. 
+
+## Selected solution for OLM-v0 catalogs
+
+This document will only focus on the feature design for OLM-v0 catalogs. Reflecting the mirrored operators in OLM-v1 catalogs shall be handled in a different feature request, at a later time.
+
+From the above solutions for OLM-v0 catalogs, the decision was made to rebuild catalog images inserting the FBC filtered by minVersion (and maxVersion) in the image. 
+
+**This solution is TEMPORARY.** It shall be accompanied by proper documentation explaining that filtered catalogs will not benefit from signature verification, and that this problem will be **fully solved by migrating to OLM-v1 catalogs.**
```

Comment:
> It shall be accompanied by proper documentation explaining that filtered catalogs will not benefit from signature verification

As in my other comment, customers can still consume these rebuilt/custom catalogs in their clusters with the signature validation enforced.  

Just that the newly rebuilt catalogs should be viewed as customers' own "custom catalogs", which means they should be instructed to sign all of their rebuilt catalog images, including those built by oc-mirror, using their own private key.

This way, they can still consume these custom catalogs in clusters enforcing the image signatures need to be either from official Red Hat or their private key.